src/Components/Web/src/Web/RenderTreeBuilderExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components.Web
+{
+    /// <summary>
+    /// Provides methods for building a collection of <see cref="RenderTreeFrame"/> entries.
+    /// </summary>
+    public static class RenderTreeBuilderExtensions
+    {
+        // The "prevent default" and "stop propagation" flags behave like attributes, in that:
```

Comment:
FYI the compiler support for this will need to do feature detection to see if these APIs are defined. Also the compiler won't likely codegen them as extension methods just because it's not something we tend to do - I'll leave that up to @ajaybhargavb though.

Reply:
@rynowak @SteveSandersonMS, do we plan to do the compiler work for this in preview2?

Reply:
3